x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -1120,6 +1121,115 @@ public void testAutoFollowSearchableSnapshotsFails() throws Exception {
+                }
+                verifyDataStream(leaderClient, dataStreamName, backingIndexName(dataStreamName, 1));
+                verifyDocuments(leaderClient, dataStreamName, numDocs);
+            }
+            assertBusy(() -> {
+                assertThat(getNumberOfSuccessfulFollowedIndices(), equalTo(initialNumberOfSuccessfulFollowedIndices + 1));
+                verifyDataStream(client(), dataStreamName, backingIndexName(dataStreamName, 1));
+                ensureYellow(dataStreamName);
+                verifyDocuments(client(), dataStreamName, numDocs);
+            });
+
+            // Rollover in leader cluster and ensure second backing index is replicated:
+            try (var leaderClient = buildLeaderClient()) {
+                var rolloverRequest = new Request("POST", "/" + dataStreamName + "/_rollover");
+                assertOK(leaderClient.performRequest(rolloverRequest));
+                verifyDataStream(leaderClient, dataStreamName, backingIndexName(dataStreamName, 1), backingIndexName(dataStreamName, 2));
+
+                var indexRequest = new Request("POST", "/" + dataStreamName + "/_doc");
+                indexRequest.addParameter("refresh", "true");
+                indexRequest.setJsonEntity("{\"@timestamp\": \"" + DATE_FORMAT.format(new Date()) + "\",\"message\":\"abc\"}");
+                assertOK(leaderClient.performRequest(indexRequest));
+                verifyDocuments(leaderClient, dataStreamName, numDocs + 1);
+            }
+            assertBusy(() -> {
+                assertThat(getNumberOfSuccessfulFollowedIndices(), equalTo(initialNumberOfSuccessfulFollowedIndices + 2));
+                verifyDataStream(client(), dataStreamName, backingIndexName(dataStreamName, 1), backingIndexName(dataStreamName, 2));
+                ensureYellow(dataStreamName);
+                verifyDocuments(client(), dataStreamName, numDocs + 1);
+            });
+
+            backingIndexNames = verifyDataStream(
+                client(),
+                dataStreamName,
+                backingIndexName(dataStreamName, 1),
+                backingIndexName(dataStreamName, 2)
+            );
+
+            // Promote local data stream
+            var promoteRequest = new Request("POST", "/_data_stream/_promote/" + dataStreamName);
+            Response response = client().performRequest(promoteRequest);
+            assertOK(response);
+        } finally {
+            if (backingIndexNames == null) {
+                // we failed to compute the actual backing index names in the test because we failed earlier on, guessing them on a
+                // best-effort basis
+                backingIndexNames = List.of(backingIndexName(dataStreamName, 1), backingIndexName(dataStreamName, 2));
+            }
+
+            cleanUpFollower(backingIndexNames, List.of(dataStreamName), List.of(autoFollowPatternName));
+            cleanUpLeader(backingIndexNames.subList(0, 1), List.of(dataStreamName), List.of());
```

Comment:
Why are we also trying to clean up the individual backing indices? Is cleaning up the data stream not enough?

Reply:
I must admit I copied this from the cleanup step of another test the creates and promotes DataStreams in this class so I'm assuming the original author had their reasons?

I think possibly this ensures the indices are cleaned up, even if the data stream gets itself into into a corrupted state during the test and doing the cleanup prevents other tests in the suite then failing?

Reply:
We all have done that :). 

Unfortunately, I also cannot say with confidence that this can never happen. 

I have to admit that I do find this way of "tear down" a bit unusual. I would prefer to have a tear down method that would wipe the cluster at the end of every test so we do not have to deal with this but I do understand that this is out of the scope of this. 

So, happy to keep this, but maybe add a comment, that we do not know if this is necessary but we do it to be sure? So, at least the next maintainer can factor this in. What do you think?

Reply:
Comment added :-)